core/scheduler/src/main/scala/org/apache/openwhisk/core/scheduler/container/CreationJobManager.scala

```diff
@@ -0,0 +1,268 @@
+import pureconfig.loadConfigOrThrow
+
+import scala.collection.concurrent.TrieMap
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+
+sealed trait CreationJob
+case class RegisterCreationJob(msg: ContainerCreationMessage) extends CreationJob
+case class FinishCreationJob(ack: ContainerCreationAckMessage) extends CreationJob
+case class ReschedulingCreationJob(tid: TransactionId,
+                                   creationId: CreationId,
+                                   invocationNamespace: String,
+                                   action: FullyQualifiedEntityName,
+                                   revision: DocRevision,
+                                   actionMetaData: WhiskActionMetaData,
+                                   schedulerHost: String,
+                                   rpcPort: Int,
+                                   retry: Int)
+    extends CreationJob {
+
+  def toCreationMessage(sid: SchedulerInstanceId, retryCount: Int): ContainerCreationMessage =
+    ContainerCreationMessage(
+      tid,
+      invocationNamespace,
+      action,
+      revision,
+      actionMetaData,
+      sid,
+      schedulerHost,
+      rpcPort,
+      retryCount,
+      creationId)
+}
+
+abstract class CreationJobState(val creationId: CreationId,
+                                val invocationNamespace: String,
+                                val action: FullyQualifiedEntityName,
+                                val revision: DocRevision)
+case class FailedCreationJob(override val creationId: CreationId,
+                             override val invocationNamespace: String,
+                             override val action: FullyQualifiedEntityName,
+                             override val revision: DocRevision,
+                             error: ContainerCreationError,
+                             message: String)
+    extends CreationJobState(creationId, invocationNamespace, action, revision)
+case class SuccessfulCreationJob(override val creationId: CreationId,
+                                 override val invocationNamespace: String,
+                                 override val action: FullyQualifiedEntityName,
+                                 override val revision: DocRevision)
+    extends CreationJobState(creationId, invocationNamespace, action, revision)
+
+case object GetPoolStatus
+
+case class JobEntry(action: FullyQualifiedEntityName, timer: Cancellable)
+
+class CreationJobManager(feedFactory: (ActorRefFactory, String, Int, Array[Byte] => Future[Unit]) => ActorRef,
+                         schedulerInstanceId: SchedulerInstanceId,
+                         dataManagementService: ActorRef)(implicit actorSystem: ActorSystem, logging: Logging)
+    extends Actor {
+  private implicit val ec: ExecutionContext = actorSystem.dispatcher
+  private val baseTimeout = loadConfigOrThrow[Int](ConfigKeys.schedulerInProgressJobRetentionSecond).seconds
```

Comment:
For time configuration, it is better to change it to `loadConfigOrThrow[FiniteDuration]`

Reply:
great catch, I will update